src/lib/aggregations/get-cardinality.spec.ts

```diff
@@ -16,5 +28,66 @@ describe('getCardinalityAggregation', () => {
         })
     })
 
-    test.todo('accepts only schema field with keyword')
+    it('should queries elasticsearch for cardinality aggregation with specific field', async () => {
+        const service = app.get(ElasticsearchService)
+
+        const result = await service.search(HomeDocument, {
+            size: 0,
+            aggregations: {
+                testAggregation: getCardinalityAggregation('address.keyword')
+            }
+        })
+
+        expect(result.aggregations.testAggregation.value).toBeDefined()
+    })
+
+    it('should queries elasticsearch for cardinality aggregation with specific field and precision control', async () => {
+        const service = app.get(ElasticsearchService)
+
+        const result = await service.search(HomeDocument, {
+            size: 0,
+            aggregations: {
+                testAggregation: getCardinalityAggregation('address.keyword', {
+                    precision_threshold: 1
+                })
+            }
+        })
+
+        expect(result.aggregations.testAggregation.value).toBeDefined()
+    })
+
+    it('should queries elasticsearch for cardinality aggregation with script', async () => {
+        const service = app.get(ElasticsearchService)
+
+        const result = await service.search(HomeDocument, {
+            size: 0,
+            aggregations: {
+                testAggregation: getCardinalityAggregation({
+                    script: `doc['address.keyword'].value`
+                })
+            }
+        })
+
+        expect(result.aggregations.testAggregation.value).toBeDefined()
+    })
+
+    it('should queries elasticsearch for cardinality aggregation with script and precision control', async () => {
```

Comment:
`with script and precision control` change to `with script and precision threshold`